src/server/analytic_services/prometheus_reporting.js

```diff
@@ -2,359 +2,423 @@
+        configuration: {
+            help: 'Object Bucket Claims'
+        }
+    }, {
+        type: 'Gauge',
+        name: 'num_objects_buckets_claims',
+        generate_default_set: true,
+        configuration: {
+            help: 'Objects On Object Bucket Claims'
+        }
+    }, {
+        type: 'Gauge',
+        name: 'reduction_ratio',
+        generate_default_set: true,
+        configuration: {
+            help: 'Object Efficiency Ratio'
+        }
+    }, {
+        type: 'Gauge',
+        name: 'object_savings_logical_size',
+        generate_default_set: true,
+        configuration: {
+            help: 'Object Savings Logical'
+        }
+    }, {
+        type: 'Gauge',
+        name: 'object_savings_physical_size',
+        generate_default_set: true,
+        configuration: {
+            help: 'Object Savings Physical'
+        }
+    }, {
+        type: 'Gauge',
+        name: 'rebuild_progress',
+        generate_default_set: true,
+        configuration: {
+            help: 'Rebuild Progress'
+        }
+    }, {
+        type: 'Gauge',
+        name: 'rebuild_time',
+        generate_default_set: true,
+        configuration: {
+            help: 'Rebuild Time'
+        }
+    }, {
+        type: 'Gauge',
+        name: 'bucket_status',
+        configuration: {
+            help: 'Bucket Health',
+            labelNames: ['bucket_name']
+        }
+    }, {
+        type: 'Gauge',
+        name: 'bucket_capacity',
+        configuration: {
+            help: 'Bucket Capacity Precent',
+            labelNames: ['bucket_name']
+        }
+    }, {
+        type: 'Gauge',
+        name: 'bucket_quota',
+        configuration: {
+            help: 'Bucket Quota Precent',
+            labelNames: ['bucket_name']
+        }
+    }, {
+        type: 'Gauge',
+        name: 'resource_status',
+        configuration: {
+            help: 'Resource Health',
+            labelNames: ['resource_name']
+        }
+    }, {
+        type: 'Gauge',
+        name: 'system_links',
+        configuration: {
+            help: 'System Links',
+            labelNames: ['resources', 'buckets', 'dashboard']
+        }
+    }, {
+        type: 'Gauge',
+        name: 'health_status',
+        generate_default_set: true,
+        configuration: {
+            help: 'Health status'
+        }
     }
+]);
 
-    export_metrics() {
-        const exported = _.map(METRIC_RECORDS, metric =>
-                this._prom_client.register.getSingleMetricAsString(metric.configuration.name))
-            .join('\n');
+class NooBaaCoreReport extends BasePrometheusReport {
+    constructor() {
+        super();
 
-        return exported;
+        this._metrics = {};
+        for (const m of NOOBAA_CORE_METRICS) {
+            this._metrics[m.name] = new prom_client[m.type]({
+                ...m.configuration,
+                name: `${this.metric_prefix}${m.name}`,
+                registers: [this._registry]
+            });
```

Comment:
Those are common functionality. Can we move those into the base class BasePrometheusReport so that we don't duplicate code in inheriting classes.

Reply:
I don't think this should be a common functionality because I would like to implement the new reports differently.
I think that trying to be "metadata-driven" in this case is a bad choice. The code of a report can be greatly simplified by just creating the metric itself in the constructor. For example for new reports I will do the following in the report constructor:
```
// Define storage for the metrics:
this._metrics = {
    // This is a dummy metric that is used as an example, Gauge is use
    // here forbut it can be any of the metric types defined in the prom
    // client lib at https://github.com/siimon/prom-client with its
    // specific configuration.
    demo_metric: new client.Gauge({
        name: this.get_prefixed_name('demo_metric'),
        registers: [this._registry],
        help: 'Description line for demo_metric',
        // Other configurqation that should apply for this metric
    }),
};
```

Again, I didn't want to touch the existing metrics.

Reply:
ok